test/functional/Java8andUp/src/org/openj9/test/nogc/TestGCPolicyNogc.java

```diff
@@ -0,0 +1,368 @@
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.openj9.test.nogc;
+
+import org.testng.AssertJUnit;
+import org.testng.annotations.Test;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.AfterTest;
+
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryPoolMXBean;
+import java.lang.management.MemoryMXBean;
+import java.lang.management.MemoryNotificationInfo;
+import java.lang.management.MemoryType;
+import java.lang.management.MemoryUsage;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import javax.management.ListenerNotFoundException;
+import javax.management.Notification;
+import javax.management.NotificationEmitter;
+import javax.management.NotificationFilter;
+import javax.management.NotificationListener;
+import javax.management.openmbean.CompositeData;
+
+import com.sun.management.GarbageCollectionNotificationInfo;
+import com.sun.management.GcInfo;
+
+@Test(groups={ "level.extended" })
+public class TestGCPolicyNogc {
+
+	static final class GCState 
+	{ 
+		final GarbageCollectorMXBean gcBean; 
+		final boolean assumeGCIsPartiallyConcurrent; 
+		final boolean assumeGCIsOldGen; 
+		long  totalCollectionCount = 0;
+		long  totalCollectionDuration = 0; 
+		long  lastCollectionID = 0;
+
+		GCState(GarbageCollectorMXBean gcBean, boolean assumeGCIsPartiallyConcurrent, boolean assumeGCIsOldGen) 
+		{ 
+			this.gcBean = gcBean; 
+			this.assumeGCIsPartiallyConcurrent = assumeGCIsPartiallyConcurrent; 
+			this.assumeGCIsOldGen = assumeGCIsOldGen; 
+		} 
+	}
+	
+	final Map<String, GCState> gcStates = new HashMap<>();
+	
+    private static final long JVM_START_TIME = ManagementFactory.getRuntimeMXBean().getStartTime();
+
+    private static DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+
+    private static final long ONE_KBYTE = 1024;
+
+	private long totalGcTimeSpend = 0;
+	private long used_CollectionThresholdExceeded = 0;
+	private long used_ThresholdExceeded = 0;
+
+	private NotificationEmitter gcEmitter;
+	private NotificationEmitter  memoryEmitter;
+	private NotificationListener gcListener;
+	private NotificationListener memoryListener;
+	
+	@Test (priority=1)
+	public final void testNogcModeEnabled() {
+		System.out.println("testNogcModeEnabled start");
```

Comment:
Its better if you use the testng logger for printing, instead of System.out.println, as we want to be able to turn off, or turn down the volume of output if necessary.  You can see how the logger is used in this [JLM_Test example](https://github.com/eclipse/openj9/blob/master/test/functional/JLM_Tests/src_80/org/openj9/test/java/lang/management/TestThreadInfo.java#L52).  In most cases, it is likely that you can use logger.info level, which prints out by default, or logger.debug level if you only want it to print if debug level is set before the run. 

This comment applies to all System.out.println statements in testng classes.

Reply:
done